transformer_engine/pytorch/transformer.py

```diff
@@ -239,13 +241,6 @@ def __init__(
     ) -> None:
         super().__init__()
 
-        if self_attn_mask_type is not None:
-            warnings.warn(
-                "Argument :attr:`self_attn_mask_type` has been moved to the `forward` method and"
-                "is deprecated. It will be fully removed in future releases.",
-                category=DeprecationWarning,
-            )
```

Comment:
We deprecated this option recently in https://github.com/NVIDIA/TransformerEngine/pull/402, so I think it's worth a proper discussion if we want to bring it back. For the record, I think this API is the right one: allow the user to specify a mask type in the constructor and to override their choice in `forward`.

Pinging @ksivaman.

Reply:
Yeah, we reverted this because onnx doesn't take dynamic string arguments, so things like the mask type or qkv format would have to be specified in the init rather than forward.